store/sqlstore/status_store.go

```diff
@@ -99,6 +102,110 @@ func (s SqlStatusStore) GetByIds(userIds []string) ([]*model.Status, error) {
 	return statuses, nil
 }
 
+// MySQL doesn't have support for RETURNING clause, so we use a transaction to get the updated rows.
+func (s SqlStatusStore) updateExpiredStatuses(t *gorp.Transaction) ([]*model.Status, error) {
+	currUnixTime := time.Now().UTC().Unix()
+	selectQuery, selectParams, _ := s.getQueryBuilder().
+		Select("UserId").
+		From("Status").
+		Where(
+			sq.And{
+				sq.Eq{"Status": model.STATUS_DND},
+				sq.Gt{"DNDEndTimeUnix": 0},
+				sq.LtOrEq{"DNDEndTimeUnix": currUnixTime},
+			},
+		).ToSql()
+	var selectedUsers []string
+	_, err := t.Select(&selectedUsers, selectQuery, selectParams...)
+	if err != nil {
+		return nil, errors.Wrap(err, "getExpiredDNDStatusesT: failed to get expired dnd statuses")
+	}
+	updateQuery, args, _ := s.getQueryBuilder().
+		Update("Status").
+		Where(
+			sq.And{
+				sq.Eq{"Status": model.STATUS_DND},
+				sq.Gt{"DNDEndTimeUnix": 0},
+				sq.LtOrEq{"DNDEndTimeUnix": currUnixTime},
+			},
+		).
+		Set("Status", sq.Expr("PrevStatus")).
+		Set("PrevStatus", model.STATUS_DND).
+		Set("DNDEndTimeUnix", 0).
+		Set("Manual", false).
+		ToSql()
+
+	if _, err := t.Exec(updateQuery, args...); err != nil {
+		return nil, errors.Wrapf(err, "failed to update statuses")
+	}
+
+	var updatedStatuses []*model.Status
+	selectQuery, selectParams, _ = s.getQueryBuilder().
```

Comment:
Why do we need to do this query? We know what are the things we are changing- status, prevStatus, dndendTimeUnix and manual. We can get all the rows from the first query, and update the records ourselves and return.